SearchTermList and Keyword Search with integration to redux

VasilyShelkov opened the pull request:
for #4

VasilyShelkov committed:
Added Search List and SearchKeywords input field

Also made the focussed placeholder color grey

VasilyShelkov committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

Conflicts:
	src/client/App.js

VasilyShelkov committed:
Added redux and searchReducer

Crazy stuff, added reducer and spec, action and connected the CurrentSearch component to the store

VasilyShelkov committed:
Removed Batch items, redux-logger and tested ADD_SEARCH_QUERY dispatch action

Decided didn't need dispatchBatchActions and removed it

Instead having a list of paramType

Added redux chrome extension if want to use instead of using middleware redux-logger

VasilyShelkov committed:
Refactored CurrentSearch, Added SearchTermsList

VasilyShelkov committed:
JSLint errors

Winwardo commented on src/client/index.js:
```diff
@@ -1,18 +1,41 @@
+    <Provider store={finalStore}>
+      <App />
+    </Provider>,
+    rootEl
```
rootElement pls

Winwardo commented on src/client/index.js:
```diff
@@ -1,18 +1,41 @@
 import 'babel-polyfill';
 import ReactDOM from 'react-dom';
 import React from 'react';
+import {combineReducers, compose, applyMiddleware, createStore } from 'redux';
```
import { c

Winwardo commented on src/client/search/Current.js:
```diff
@@ -0,0 +1,60 @@
+
+let nextSearchTermId = 0;
+let Current = ({ dispatch }) => {
+	let showSearchkeyword = false;
```
spaces not tabs

Winwardo commented on src/client/search/Current.js:
```diff
@@ -0,0 +1,60 @@
+import SearchTermsList from './SearchTermsList';
+
+let nextSearchTermId = 0;
+let Current = ({ dispatch }) => {
```
Current what?

Winwardo commented on src/client/search/Current.js:
```diff
@@ -0,0 +1,60 @@
+			<div style={{cursor: 'text'}} onClick={() => {
+				showSearchkeyword = !showSearchkeyword;
+				if(showSearchkeyword) {
+					$("#searchKeywordContainer").slideDown("fast", () => {
```
Single quotes instead of double - JSCS should have caught this?? :s

Winwardo commented on src/client/search/SearchActions.spec.js:
```diff
@@ -0,0 +1,52 @@
+    const expectedQuery = 'Footy';
+    const expectedAction = {
+      type: 'ADD_SEARCH_TERM',
+      id,
```
I don't get what this does? o.O
Where's the colon?

VasilyShelkov replied:
This is es6 syntax turns into:
const expectedAction = {
  type: 'ADD_SEARCH_TERM',
  id: id //the id const declared at the top, just shorthand to write id
  ...
};

Winwardo replied:
Magical

Winwardo commented on src/client/search/SearchTermsList.js:
```diff
@@ -0,0 +1,35 @@
+      {searchTerms.map(term => {
+        return (
+          <TermItem
+						key={term.id}
```
spaces :P

VasilyShelkov replied:
what's the issue with this ?

Winwardo commented on src/client/search/SearchTermsReducer.js:
```diff
@@ -0,0 +1,27 @@
+const SearchTermsReducer = (state = [], action) => {
```
SearchTermListReducer would be more readable

VasilyShelkov replied:
It's not got to do with a list component though..I've deliberately not named it with List since that's common practice with reducers.

Winwardo commented on src/client/search/SearchTermsReducer.js:
```diff
@@ -0,0 +1,27 @@
+      ...state,
+      SearchTermReducer(undefined, action),
+			];
+    default:
```
Curly braces even for single line statements please, easy to add a debugging "console.log" or whatever and mess this up

VasilyShelkov replied:
I don't understandwhere do you want curly braces ? 

you shouldn't be debugging a reducer from here anyway... you should either have tests or be using the redux debug tool

Winwardo commented on src/client/search/SearchTermsReducer.spec.js:
```diff
@@ -0,0 +1,63 @@
+      source: 'twitter',
+    },];
+
+    deepFreeze(stateBefore);
```
I get what this is doing, but is it totally necessary for the test?
Although tbf in another language the objects would be immutable before hand anyway

VasilyShelkov replied:
yes, since it's key when using Redux that the state nor action is mutated, again common practise... if it gets mutated will cause problems in the background

VasilyShelkov committed:
Pull Request comment Fixes

-Renamed Current = CurrentQueryTerms
-changed Rootel to rootelement in index.js
-Fixed spaces
-jcsc added trailing commas in test

Winwardo commented on src/client/search/Current.js:
```diff
@@ -0,0 +1,60 @@
+	let searchKeyword;
+	let searchKeywordContainer;
+	return(
+		<div className="row ui raised segment">
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2183766983/issues/source?bid=3148782&fileBranchId=3165072#l12)

Winwardo commented on src/client/search/CurrentQueryTerms.js:
```diff
@@ -0,0 +1,56 @@
+  let showSearchkeyword = false;
+  let searchKeyword;
+  return (
+    <div className="row ui raised segment">
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2183790101/issues/source?bid=3148782&fileBranchId=3165072#l11)

VasilyShelkov commented:
@Winwardo, ready for further comments and merge, made changes from previous comments...not sure why it's picked up the mixed double sand single quotes only in that file, when it has it in the rest of the files... it's consistent:
html props uses double quotes vs any javascript uses single quotes
